crates/red_knot_python_semantic/src/types.rs

```diff
@@ -2361,7 +2360,7 @@ impl<'db> KnownClass {
         }
     }
 
-    pub fn try_from_file(db: &dyn Db, file: File, class_name: &str) -> Option<Self> {
+    pub fn try_from_file_and_symbol(db: &dyn Db, file: File, class_name: &str) -> Option<Self> {
```

Comment:
nit: I would expect a `Symbol` or similar to be passed in, based on this name; I think `try_from_file_and_name` would be clearer